pandapipes/converter/stanet/data_cleaning.py

```diff
@@ -0,0 +1,118 @@
+import copy
+import os
+
+import numpy as np
+from pandapipes.converter.stanet.preparing_steps import connection_pipe_section_table
+
+
+def sort_by_pos(group):
+    group = group.sort_values("posL")
+    group_characteristics(group)
+    return group
+
+
+def sort_by_flow(group):
+    group["follower"] = group.FLUSSB.apply(lambda b: group.index[group.FLUSSA == b][0]
+                                           if b in group.FLUSSA.values else np.NaN)
+    ls = list(group.loc[~group.FLUSSA.isin(group.FLUSSB)].index)
+    assert len(ls) == 1
+    while len(ls) < len(group):
+        ls.append(int(group.follower[ls[-1]]))
+    group = group.loc[ls]
+    group = group_characteristics(group)
+    assert np.allclose(group.posL.values, np.sort(group.posL.values))
+
+    return group
+
+def group_characteristics(group):
+    group["nodes_in_group"] = len(group)
+    group["current_node"] = np.arange(1, len(group) + 1)
+    pos_l_previous = np.zeros(len(group))
+    pos_l_previous[1:] = group["posL"].values[:-1]
+    group["len_previous"] = group["posL"] - pos_l_previous
+    group["len_remaining"] = group["RORL"] - group["posL"]
+    return group
+
+
+def remove_connections_zero_length(stored_data, cutoff_length=1e-1, save_path=None,
+                                   write_header=False, save_excel=False):
+    connections = stored_data["house_connections"].copy()
+    used_connections = np.isin(connections.STANETID, stored_data["house_pipes"].CLIENTID) \
+        | np.isin(connections.STANETID, stored_data["house_pipes"].CLIENT2ID)
+    unused_ids = connections.STANETID.loc[~used_connections].values
+
+    cons = connection_pipe_section_table(stored_data, True, False)
```

Comment:
the function has four inputs. One is missing. @dlohmeier, do you know, which one?

Reply:
I corrected this.